R-package/vignettes/MultidimLstm.Rmd

```diff
@@ -0,0 +1,401 @@
+    batch.size = batch.size, shuffle = TRUE)
+
+eval.data <- mx.io.arrayiter(data = trainX[, , eval_ids, drop = F], label = trainY[, eval_ids],
+    batch.size = batch.size, shuffle = FALSE)
+
+## Create the symbol for RNN
+symbol <- rnn.graph(num_rnn_layer = 1, num_hidden = 5, input_size = NULL, num_embed = NULL, num_decode = 1,
+    masking = F, loss_output = "linear", dropout = 0.2, ignore_label = -1, cell_type = "lstm", output_last_state = T,
+    config = "one-to-one")
+
+mx.metric.mse.seq <- mx.metric.custom("MSE", function(label, pred) {
+    label = mx.nd.reshape(label, shape = -1)
+    pred = mx.nd.reshape(pred, shape = -1)
+    res <- mx.nd.mean(mx.nd.square(label - pred))
+    return(as.array(res))
+})
+
+
+
+ctx <- mx.cpu()
+
+initializer <- mx.init.Xavier(rnd_type = "gaussian", factor_type = "avg", magnitude = 3)
+
+optimizer <- mx.opt.create("adadelta", rho = 0.9, eps = 1e-05, wd = 1e-06, clip_gradient = 1, rescale.grad = 1/batch.size)
+
+logger <- mx.metric.logger()
+epoch.end.callback <- mx.callback.log.train.metric(period = 10, logger = logger)
+
+## train the network
+system.time(model <- mx.model.buckets(symbol = symbol, train.data = train.data, eval.data = eval.data,
+    num.round = 100, ctx = ctx, verbose = TRUE, metric = mx.metric.mse.seq, initializer = initializer,
+    optimizer = optimizer, batch.end.callback = NULL, epoch.end.callback = epoch.end.callback))
+ctx <- mx.cpu()
+```
+Output:
+```
+Start training with 1 devices
+[1] Train-MSE=0.197570244409144
+[1] Validation-MSE=0.0153861071448773
+[2] Train-MSE=0.0152517843060195
+[2] Validation-MSE=0.0128299412317574
+[3] Train-MSE=0.0124418652616441
+[3] Validation-MSE=0.010827143676579
+[4] Train-MSE=0.0105128229130059
+[4] Validation-MSE=0.00940261723008007
+[5] Train-MSE=0.00914482437074184
+[5] Validation-MSE=0.00830172537826002
+[6] Train-MSE=0.00813581114634871
+[6] Validation-MSE=0.00747016374953091
+[7] Train-MSE=0.00735094994306564
+[7] Validation-MSE=0.00679832429159433
+[8] Train-MSE=0.00672049634158611
+[8] Validation-MSE=0.00623159145470709
+[9] Train-MSE=0.00620287149213254
+[9] Validation-MSE=0.00577476259786636
+[10] Train-MSE=0.00577280316501856
+[10] Validation-MSE=0.00539038667920977
+[11] Train-MSE=0.00540679777041078
+[11] Validation-MSE=0.00506085657980293
+[12] Train-MSE=0.0050867410376668
+[12] Validation-MSE=0.00477395416237414
+[13] Train-MSE=0.00480019277893007
+[13] Validation-MSE=0.00450056773843244
+[14] Train-MSE=0.00453343892004341
+[14] Validation-MSE=0.00424888811539859
+[15] Train-MSE=0.00428280527703464
+[15] Validation-MSE=0.00400642631575465
+[16] Train-MSE=0.00405749503988773
+[16] Validation-MSE=0.00380465737544
+[17] Train-MSE=0.00386031914968044
+[17] Validation-MSE=0.00360809749690816
+[18] Train-MSE=0.00368094681762159
+[18] Validation-MSE=0.00342673255363479
+[19] Train-MSE=0.00352097053546459
+[19] Validation-MSE=0.00327468500472605
+[20] Train-MSE=0.0033796411473304
```

Comment:
Let's truncate the middle part of the output.

Reply:
done